BetaCameras/RealSense2/RealSense2API.cs

```diff
@@ -0,0 +1,596 @@
+            int enabled = 0;
+            rs2_is_enabled(device.ptr, ref enabled, &error);
+            HandleError(error);
+
+            return enabled == 0 ? false : true;
+        }
+
+        unsafe public static void EnabledAdvancedMode(RS2Device device, bool enable)
+        {
+            IntPtr error = IntPtr.Zero;
+            rs2_toggle_advanced_mode(device.ptr, enable ? 1 : 0, &error);
+            HandleError(error);
+        }
+
+        unsafe public static void LoadAdvancedConfig(string config, RS2Device device)
+        {
+            IntPtr error = IntPtr.Zero;
+            rs2_load_json(device.ptr, config, (uint)config.ToCharArray().Length, &error);
+            HandleError(error);
+        }
+
+        unsafe public static Intrinsics GetIntrinsics(RS2StreamProfile profile)
+        {
+            IntPtr error = IntPtr.Zero;
+            Intrinsics intrinsics = new Intrinsics();
+            rs2_get_video_stream_intrinsics(profile.ptr, &intrinsics, &error);
+            HandleError(error);
+
+            return intrinsics;
+        }
+
+        unsafe public static Extrinsics GetExtrinsics(RS2StreamProfile from, RS2StreamProfile to)
+        {
+            IntPtr error = IntPtr.Zero;
+            Extrinsics extrinsics = new Extrinsics();
+            rs2_get_extrinsics(from.ptr, to.ptr, &extrinsics, &error);
+            HandleError(error);
+
+            return extrinsics;
+        }
+
+        unsafe public static float GetDepthScale(RS2Pipeline pipe)
+        {
+            IntPtr error = IntPtr.Zero;
+            float res = 0.0f;
+
+            IntPtr profile = rs2_pipeline_get_active_profile(pipe.ptr, &error);
+            HandleError(error);
+            IntPtr device = rs2_pipeline_profile_get_device(profile, &error);
+            HandleError(error);
+            IntPtr list = rs2_query_sensors(device, &error);
+            HandleError(error);
+            int sensorCount = rs2_get_sensors_count(list, &error);
+            HandleError(error);
+
+            for (int i = 0; i < sensorCount; i++)
+            {
+                IntPtr sensor = rs2_create_sensor(list, i, &error);
+                HandleError(error);
+
+                float scale = rs2_get_depth_scale(sensor, &error);
+                HandleError(error);
+
+                char* info = rs2_get_sensor_info(sensor, CameraInfo.NAME, &error);
+                string infoString = Marshal.PtrToStringAnsi((IntPtr)info);
+                HandleError(error);
+
+                rs2_delete_sensor(sensor);
+
+                if (infoString == "Stereo Module")
+                {
+                    res = scale;
+                    break;
+                }
+            }
+
+            rs2_delete_sensor_list(list);
+            rs2_delete_device(device);
+            rs2_delete_pipeline_profile(profile);
+
+            return res;
+        }
+
+        unsafe private static void HandleError(IntPtr e)
+        {
+            if (e != IntPtr.Zero)
```

Comment:
I would check if `e == IntPtr.Zero` and then return. That way the rest of the method is less indented. When the nesting gets deeper this really helps readability.

Reply:
fixed